src/routes/(app)/form/[id]/+page.server.ts

```diff
@@ -47,14 +48,26 @@ export const actions = {
 		const htmlContent = `Message envoyé par formulaire sur le site ${ORIGIN}<br> Adresse de contact : ${
 			parsed.from ? parsed.from : 'non-renseigné'
 		}  <br> Message envoyé :<br> ${parsedMessageHtml}`;
-		await collections.emailNotifications.insertOne({
-			_id: new ObjectId(),
-			createdAt: new Date(),
-			updatedAt: new Date(),
-			subject: parsed.subject,
-			htmlContent: htmlContent,
-			dest: parsed.target || SMTP_USER
-		});
+		if (parsed.target.startsWith('npub')) {
+			await collections.nostrNotifications.insertOne({
+				_id: new ObjectId(),
+				createdAt: new Date(),
+				kind: Kind.EncryptedDirectMessage,
+				updatedAt: new Date(),
+				content: htmlContent,
+				dest: parsed.target
+			});
+		} else {
+			await collections.emailNotifications.insertOne({
+				_id: new ObjectId(),
+				createdAt: new Date(),
+				updatedAt: new Date(),
+				subject: parsed.subject,
+				htmlContent: htmlContent,
+				dest: parsed.target || SMTP_USER
```

Comment:
I have mixed feeling about it.
Let's take a live example : on PVH Editions pilot, we have
- contact address for EU/FR
- contact address for CH
- contact address for book submission
- contact address for logistic
- contact address for partnership
So, by email, we can define a target.
So, same thing for Nostr.
And, clearly, it can become abusive, as you said.
We can imagine a future feature (like, future, in the next days) that requires validation from the target address (nostr or email) before sending stuff.
If it seems good to you, we can merge as it is and I can create the ticket right after your greenlight.

Reply:
no you can just remove this @ithiame :

https://github.com/B2Bitcoin/beBOP/blob/20c905d55a297e6e368a9049a4ec213989ee7f3e/src/lib/components/ContactForm.svelte#L25

and in the backend load the target from DB (from the contact form in DB).

no need for anything more complicated, just do not load the target from user input

Reply:
Oh, you meant that a normal, non-employee visitor, was able to contact anyone through a webform ? Indeed, that's not supposed to happen.